conjure-runtime/src/client_factory.rs

```diff
@@ -401,28 +414,66 @@ impl ClientFactory {
                 move |c| c.merged_service(&service).unwrap_or_default()
             })
             .try_map({
+                let mut client_handles = HashMap::<Host, ClientHandle>::new();
                 move |c| {
+                    let mut new_client_handles = HashMap::new();
                     let mut clients = HashMap::new();
+                    let mut errors = vec![];
 
                     for (i, uri) in c.uris().iter().enumerate() {
                         let host = Host { uri: uri.clone() };
                         let host_config = service_config::Builder::from(c.clone())
                             .uris([uri.clone()])
                             .build();
 
-                        let state = state_builder.build(&host_config, Some(i))?;
-                        let state = Arc::new(ArcSwap::new(state));
-                        let client = make_client(Client::new(state, None));
-
-                        clients.insert(host, client);
+                        let client_handle = match client_handles.remove(&host) {
+                            Some(mut client_handle) => {
+                                if let Err(es) = client_handle.handle.refresh(host_config) {
+                                    errors.extend(es);
+                                }
+                                client_handle
+                            }
+                            None => {
+                                let (host_config, handle) = Refreshable::new(host_config);
+                                let client = match Self::raw_client(
+                                    state_builder.clone(),
+                                    host_config,
+                                    Some(i),
+                                ) {
+                                    Ok(state) => state,
+                                    Err(e) => {
+                                        errors.push(e);
+                                        continue;
+                                    }
+                                };
+                                ClientHandle { client, handle }
+                            }
+                        };
+
+                        clients.insert(host.clone(), make_client(client_handle.client.clone()));
+                        new_client_handles.insert(host, client_handle);
                     }
 
-                    Ok(PerHostClients { clients })
+                    client_handles = new_client_handles;
```

Comment:
Where is `client_handles` used?

Reply:
It's used up on line 429.

Reply:
I think where I'm confused is this (line 457) is outside of the loop. So how does our reassignment here affect line 429?

Reply:
Oh yeah, it's not updating it for the next loop iteration, it's updating it for the next call to this closure (i.e. the next time the service config changes)